pkg/pillar/cmd/zedrouter/patchenvelopes.go

```diff
@@ -299,19 +270,12 @@ func (pes *PatchEnvelopes) UpdateEnvelopes(peInfo []types.PatchEnvelopeInfo) {
 		envelopes.Store(peUUID, pe)
 	}
 
-	toUpdate := envelopes.Keys()
-
-	toDelete, _ := generics.DiffSets(before, toUpdate)
-
-	pes.envelopes = envelopes
-
+	toDelete, _ := generics.DiffSets(before, envelopes.Keys())
 	for _, deleteUUID := range toDelete {
 		pes.envelopesToDelete.Store(deleteUUID, true)
 	}
 
-	for _, updateUUID := range toUpdate {
-		pes.envelopesToUpdate.Store(updateUUID, true)
-	}
+	pes.envelopes = envelopes
```

Comment:
This one has to be guarded by a mutex.

Reply:
It's already guarded in library [here](https://github.com/lf-edge/eve/blob/985c3dbc7320b26f9e758a8299f5af3f3815ed64/pkg/pillar/utils/generics/lockmap.go#L51)

Reply:
huh, I wonder if that was the problem all along...

Reply:
created RWMutex, running test with race didn't give me any error

Reply:
resolved